modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C

```diff
@@ -172,6 +186,10 @@ FlexiblePatternGenerator::FlexiblePatternGenerator(const InputParameters & param
     _positions.push_back(std::make_pair(extra_positions[i], extra_positions_mg_indices[i]));
   }
 
+  if (_background_subdomain_name.size() && _background_subdomain_id == Moose::INVALID_BLOCK_ID)
+    paramError("background_subdomain_id",
+               "This parameter must be provided if background_subdomain_name is provided.");
```

Comment:
we try on the other generators to let the users specify the name but only optionally the id. That way the system can pick a good id that does not conflict

Reply:
and this is why you have to hack the renameBlockGenerator a little bit to impose both an id and a name

Reply:
Automatically assign a block id for a user specified block name in this case would easily lead to conflicts if a user is using many `FlexiblePatternGenerator`s together. So I would like to ask the user to specify the id here.

Reply:
makes sense